src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.TestFramework/KeyVaultTestFixture.cs

```diff
@@ -159,12 +169,39 @@ private bool FromConfiguration()
             return result.AccessToken;
         }
 
+        public async Task<string> GetUserAccessToken(string authority, string resource, string scope)
+        {
+            string clientId = TestConfigurationManager.TryGetEnvironmentOrAppSetting("NativeClientId");
+            var context = new AuthenticationContext(authority, tokenCache);
+            if (_deviceCodeForStorageTests == null)
+            {
+                _deviceCodeForStorageTests =
+                    await context.AcquireDeviceCodeAsync(resource, clientId).ConfigureAwait(false);
+
+                Debug.WriteLine("############################################################################################");
+                Debug.WriteLine("Test won't run until you perform following steps:");
+                Debug.WriteLine($"1. Go to following url: {_deviceCodeForStorageTests.VerificationUrl}.");
+                Debug.WriteLine($"2. Insert following User Code: {_deviceCodeForStorageTests.UserCode}.");
+                Debug.WriteLine("3. Login with your username and password credentials.");
+                Debug.WriteLine("############################################################################################");
+            }
+
+            var result = await context.AcquireTokenByDeviceCodeAsync(_deviceCodeForStorageTests).ConfigureAwait(false);
+            return result.AccessToken;
+        }
+
         public KeyVaultClient CreateKeyVaultClient()
         {
             var myclient = new KeyVaultClient(new TestKeyVaultCredential(GetAccessToken), GetHandlers());
             return myclient;
         }
 
+        public KeyVaultClient CreateKeyVaultUserClient()
```

Comment:
the new method uses GetUserAccessToken instead of GetAccessToken, do you have something else in mind?